client/data/newsletter-categories/test/use-mark-as-newsletter-category-mutation.test.tsx

```diff
@@ -112,4 +113,73 @@ describe( 'useMarkAsNewsletterCategoryMutation', () => {
 
 		console.error = consoleError;
 	} );
+
+	it( 'should update optimistically newsletter categories cache', async () => {
+		( request as jest.MockedFunction< typeof request > ).mockResolvedValue( {
+			success: true,
+		} );
+
+		const mockedCategories = [
+			{
+				id: 200,
+				count: 15,
+				description: 'Fascinating art styles',
+				link: 'https://mocksite.wordpress.com/category/art-styles/',
+				name: 'Creative Expressions',
+				slug: 'creative-expressions',
+				taxonomy: 'category',
+				parent: 0,
+				meta: [ { key: 'theme', value: 'imagination' } ],
+				_links: {},
+			},
+			{
+				id: 10,
+				count: 7,
+				description: 'Exploring the unknown',
+				link: 'https://mocksite.wordpress.com/category/adventures/',
+				name: 'Adventures Beyond',
+				slug: 'adventures-beyond',
+				taxonomy: 'category',
+				parent: 0,
+				meta: [ { key: 'discovery', value: 'thrills' } ],
+				_links: {},
+			},
+		];
+
+		queryClient.setQueryData( [ 'categories', siteId ], mockedCategories );
+
+		const { result, waitFor } = renderHook( () => useMarkAsNewsletterCategoryMutation( siteId ), {
+			wrapper,
+		} );
+
+		await act( async () => {
+			await result.current.mutateAsync( 200 );
+		} );
+
+		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );
```

Comment:
You are right! In this case, it doesn't matter because the real query to retrieve the data from the backend is not there, but you are right; this is not the correct thing to wait.